pkg/oktaapi/client_wrapper.go

```diff
@@ -140,6 +141,10 @@ const euaADSourceType = "EUA-AD"
 func (cw *ClientWrapper) SearchCommonNameContains(ctx context.Context, searchTerm string) ([]*models.UserInfo, error) {
 	logger := appcontext.ZLogger(ctx)
 
+	// Sanitize searchTerm for non-word characters
+	specialCharsAndNumsRegex := regexp.MustCompile(`[^\w\s]`)
```

Comment:
I'm a _little_ worried that this regex might be too restrictive; it'd be good to make sure we can search for names with punctuation, like "O'Brian". It'd also filter out non-ASCII characters; the [`regexp` package's docs](https://pkg.go.dev/regexp) link to https://github.com/google/re2/wiki/Syntax for the syntax it uses, and it mentions `\w` is ASCII-only and that it's equivalent to `[0-9A-Za-z_]`. I'm not quite sure what a regex that accounts for Unicode characters would be, though. If Okta has documentation on an exact list of characters to filter out, or how to escape/sanitize characters that would mess up the search syntax, that might be a better approach.

Reply:
I was about to say the same. I was able to confirm a few cases that this would (as-is) filter out.
![Screenshot 2023-08-10 at 3 36 00 PM](https://github.com/CMSgov/easi-app/assets/15203744/d7c33d3a-ee06-4610-9abe-41987ba34281)
![Screenshot 2023-08-10 at 3 34 47 PM](https://github.com/CMSgov/easi-app/assets/15203744/909de621-524b-45c1-adff-ae5378e682d4)

Reply:
If we're able to catch the specific error from Okta that complains about special characters (does it come with a messages? HTTP Status code?) maybe we could swallow those and return an empty slice of results in those cases, that way we don't have to specifically try to match Okta's behavior, but just handle it when we see it

Reply:
The error text in Splunk was "the API returned an error: Invalid search criteria.. Causes: errorSummary: Invalid search syntax." Not sure what HTTP status code Okta returned.

Reply:
What I can find in the Okta docs:

From https://developer.okta.com/docs/reference/core-okta-api/#filter:
> The literal values can be strings enclosed in double quotes, numbers, date times enclosed in double quotes, and Boolean values (for example, true or false). **String literals must be valid JSON strings.**
Our code encloses the search terms in double quotes, `fmt.Sprintf(`(profile.firstName sw "%v" or profile.lastName sw "%v" or profile.displayName sw "%v")`, searchTerm, searchTerm, searchTerm)`.

From https://developer.okta.com/docs/reference/user-query/#url-encoding:
> The search parameter requires URL encoding for expressions that include characters such as spaces and double quotes used for strings. See [URL encoding](http://en.wikipedia.org/wiki/Percent-encoding). For example: `search=profile.department eq "Engineering"` is encoded as `search=profile.department%20eq%20%22Engineering%22`.
I think the Go Okta SDK already does this for us, because we use quotes in our search string without URL-encoding them.

The same page also mentions:
> Note: If you use the special character `"` within a quoted string, you must also escape it `\` and encode it. For example, `search=profile.lastName eq "bob"smith"` is encoded as `search=profile.lastName%20eq%20%22bob%5C%22smith%22`.

Reply:
If it's only `\`'s that cause issues (would be worth testing), we might be able to get away with:
* Remove any `\` from the input
* Escape any `"` in the input by adding a `\` in the front
* Passing the result to our existing code.

Reply:
@mynar7 Nice find on the more detailed API docs. https://developer.okta.com/docs/reference/api/users/#list-users-with-search is what's applicable, but the only notes it has are identical to what I quoted above about URL encoding and using `"` inside a quoted string.

I definitely see errors coming from Okta when testing in impl and using `\` in the search, as well as when I use '"'. I _don't_ see errors when searching with `/` or `?`. Furthermore, if I put `\"` in the search, I _don't_ get an Okta error. I think what's happening is that the search we call Okta with is only invalid when there's a `\` in it that's not escaping a `"` or when there's an unescaped `"`, so if we take care of those two cases, we should be fine.

Reply:
After testing this locally with GraphQL and front-end in dev, I've noted the only special characters that seem to throw the `the API returned an error: Invalid search criteria.. Causes: errorSummary: Invalid search syntax.` error are `\` and `"`.

I've attempted to URL encode the Okta query to varying degrees after reading that documentation note, but this either errors or returns no results.

I think the best solution might be to just strip `"` and `\` from queries:

```go
filterRegex := regexp.MustCompile(`[\\"]`)
searchTerm = filterRegex.ReplaceAllString(searchTerm, "")
``` 

As a side note, when testing through the front-end in dev, the GraphQL client seems to escape special characters already when sending JSON in the GQL query, so I don't think there's any front-end work necessary here.

Reply:
* Just removing those two characters sounds good. Removing `"` instead of escaping it should be fine, that probably shouldn't come up in anyone's name.
* As I mentioned above - I don't think any URL encoding's necessary in our code, I'm pretty sure the Okta SDK does that.
* Agreed that no frontend work is necessary.

Reply:
I also suspected the SDK is already encoding, but I figured it was worth a shot testing it regardless. I'll go ahead and make a commit for the above.